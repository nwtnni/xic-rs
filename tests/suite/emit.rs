#[test_generator::test_resources("tests/execute/*.xi")]
pub fn interpret(path: &str) -> anyhow::Result<()> {
    let hir = super::emit_hir(path)?;
    let lir = hir.map_ref(xic::api::emit_lir);

    let hir_stdout = super::interpret_hir(&hir)?;
    let lir_stdout = super::interpret_lir(&lir)?;

    pretty_assertions::assert_eq!(hir_stdout, lir_stdout);
    insta::assert_snapshot!(path, lir_stdout);
    Ok(())
}

#[test_generator::test_resources("tests/execute/*.xi")]
pub fn reorder(path: &str) -> anyhow::Result<()> {
    let expected_stdout = super::execute_expected(path)?;

    let reordered = super::reorder(path)?;
    let reordered_stdout = super::interpret_lir(&reordered)?;

    let cleaned = reordered
        .map(xic::api::construct_cfg)
        .map_mut(xic::api::clean_cfg)
        .map(xic::api::destruct_cfg);
    let cleaned_stdout = super::interpret_lir(&cleaned)?;

    pretty_assertions::assert_eq!(expected_stdout, reordered_stdout);
    pretty_assertions::assert_eq!(expected_stdout, cleaned_stdout);
    Ok(())
}
