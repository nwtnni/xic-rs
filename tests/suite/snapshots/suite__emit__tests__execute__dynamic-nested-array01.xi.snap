---
source: tests/suite/emit.rs
assertion_line: 31
expression: lir_stdout
---
1, 2, 3, 4, 5
2, 4, 6, 8, 10
3, 6, 9, 12, 15

